app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -151,21 +211,96 @@ fun PocketStories(
                         ListItemTabLargePlaceholder(stringResource(R.string.pocket_stories_placeholder_text)) {
                             onDiscoverMoreClicked("https://getpocket.com/explore?$POCKET_FEATURE_UTM_KEY_VALUE")
                         }
-                    } else {
+                    } else if (story is PocketRecommendedStory) {
                         PocketStory(story) {
                             val uri = Uri.parse(story.url)
                                 .buildUpon()
                                 .appendQueryParameter(URI_PARAM_UTM_KEY, POCKET_STORIES_UTM_VALUE)
                                 .build().toString()
                             onStoryClicked(it.copy(url = uri), rowIndex to columnIndex)
                         }
+                    } else if (story is PocketSponsoredStory) {
+                        Box(
+                            modifier = Modifier.onShown(0.5f) {
+                                onStoryShown(story)
+                            }
+                        ) {
+                            PocketSponsoredStory(story) {
+                                onStoryClicked(story, rowIndex to columnIndex)
+                            }
+                        }
                     }
                 }
             }
         }
     }
 }
 
+/**
+ * Add a callback for when this Composable is "shown" on the screen.
+ * This checks whether the composable has at least [threshold] ratio of it's total area drawn inside
+ * the screen bounds.
+ * Does not account for other Views / Windows covering it.
+ */
+private fun Modifier.onShown(
+    @FloatRange(from = 0.0, to = 1.0) threshold: Float,
+    onVisible: () -> Unit,
+): Modifier {
```

Comment:
Based on https://github.com/mozilla-mobile/fenix/pull/25206#discussion_r870580603 I could extract this to the `Modifier` file but not sure how to cleanly pass the toolbar height.
Adding two new parameters like `topPaddingToIgnore` / `bottomPaddingToIgnore` seems a bit forced for a scenario common to homescreen but not in other places where other Views may appear on top of a composable and obstruct it.
Ideally this shouldn't be needed but I haven't found a way for the composable to know on it's own about the toolbar appearing above it.

Reply:
One option could be instead to do something like adding a boolean instead to  `Modifier.onShown()` (something like `toolbarPresent` or `calculateWithToolbar`) to indicate if the Toolbar is present or not and then do the calculations as normal, rather than including specific padding values to calculate against.

That being said, we could totally just let this live here until it's needed somewhere else to give it time for a solution to bake.